client/src/components/QA/QaAddQuestionModal.jsx

```diff
@@ -1,18 +1,71 @@
 import React from 'react';
 import {useState, useEffect} from 'react';
+import axios from 'axios';
 
+const QaAddQuestionModal = ({setQuestionModal, product_id}) => {
+  console.log('load props product_id', typeof Number(product_id))
 
-const QaAddQuestionModal = ({setQuestionModal}) => {
 
   const [questionInput, setQuestionInput] = useState('')
   const [nicknameInput, setNicknameInput] = useState('')
   const [emailInput, setEmailInput] = useState('')
 
+  const createNewQuestion = (e) => {
 
+    if (questionInput.length < 15) {
+      alert('Please write a longer question with a minimum of 15 characters.')
+    } else if (nicknameInput.includes('@')) {
+      alert('You cannot use an email address as your nickname.')
+    } else if(nicknameInput.length < 5) {
+      alert('Please write a longer nickname with a minimum of 5 characters.')
+    } else if (emailInput.indexOf('@') === -1 || emailInput.indexOf('.') === -1) {
+      alert('Please ensure you are entering a correct email address with an "." and a "@".')
+    } else if (emailInput.length <= 5) {
+      alert('Please enter a email address.')
+    } else {
+      console.log(questionInput, nicknameInput, emailInput)
+
+      var data = JSON.stringify({
+        "body": questionInput,
+        "name": nicknameInput,
+        "email": emailInput,
+        "product_id": Number(product_id)
+      });
+
+      var config = {
```

Comment:
Instead of writing this function here why not write it in helpers/apiCalls and then import that file.